### PR TITLE
Tally yn bug

### PR DESCRIPTION
**Pull request by smharper**
Was looking through #361 and saw this bug fix (thanks for the sharp eye @nelsonag).  I thought it would be a good idea to pull that fix out and make it a separate PR.

This bug occurs with scatter_yn tallies when a neutron doesn't undergo a scattering event.  The `general_score` is called, and it will recognize that the last event wasn't a scatter so it will try to skip over the scatter_yn indices in the score array.  But it skips over the wrong number of indices.

I've added a test that will catch bugs like this in the future.

### CODE REVIEW

File: tests/test_many_scores/settings.xml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+<?xml version="1.0"?>
+<settings>
+
+  <eigenvalue>
+    <batches>10</batches>
+    <inactive>5</inactive>
+    <particles>100</particles>
```
Comment (paulromano) [inline tests/test_many_scores/settings.xml]: Sometimes low particle counts can cause problems on tests using both OpenMP and MPI, i.e. too many fission sites banked for a single thread. Can we raise the number of particles a bit, maybe 500? and then just reduce the number of batches.

File: tests/test_many_scores/test_many_scores.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+def test_output_exists():
+    assert os.path.exists(os.path.join(cwd, 'tallies.out')), 'Tally output file does not exist.'
+
+def test_results():
+    statepoint = glob.glob(os.path.join(cwd, 'statepoint.10.*'))
+    call([sys.executable, 'results.py', statepoint[0]])
+    compare = filecmp.cmp('results_test.dat', 'results_true.dat')
+    if not compare:
+      os.rename('results_test.dat', 'results_error.dat')
```
Comment (paulromano) [inline tests/test_many_scores/test_many_scores.py]: Should be indented two more spaces. I realize this is copy/paste, so the wrong indentation is elsewhere too.

File: tests/test_many_scores/test_many_scores.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    # test for openmc executable
+    if opts.exe is None:
+        raise Exception('Must specify OpenMC executable from command line with --exe.')
+
+    # run tests
+    test_run()
+    test_created_statepoint()
+    test_output_exists()
+    test_results()
+    teardown()
```
Comment (paulromano) [inline tests/test_many_scores/test_many_scores.py]: These should all be within a try/finally, with only the teardown in the finally block.

**Comment by paulromano** [comment]
Looks good to me other than those trivial comments! @nelsonag Mind taking a quick look too to make sure this is ok?

**Comment by smharper** [comment]
I fixed all of your comments for this test.  As you said, these errors exist in other tests, but I'd like to fix those later in a separate PR if you don't mind.

**Comment by paulromano** [comment]
Looks like the many_scores test fails now. Doesn't look like a random failure either.

**Comment by nelsonag** [comment]
Looks good to me!

I can't find any specifics in my commit logs, but I remember when I found this and at that point saw it wasn't in upstream, I remember being very confused but moving along with life.  I think I blamed it on git in my head, but it was probably me.  Having said that, this recollection could be completely wrong, it was almost a year ago and I'm getting old :-) Anyways, sorry about not pushing upstream if my recollection is way off

**Comment by smharper** [comment]
Whoops.  Looks like I updated the test with the wrong binary.  It should pass now.

@nelsonag, No worries.  I'm glad you found it at all.  I hope I will be able to help you with your PR over the course of the summer as well.

**Comment by paulromano** [comment]
Thanks @smharper and @nelsonag!

**Pull request merged**